tianshou/policy/modelfree/pg.py

```diff
@@ -190,18 +190,19 @@ def forward(
         """
         # TODO: rename? It's not really logits and there are particular
         #  assumptions about the order of the output and on distribution type
-        logits, hidden = self.actor(batch.obs, state=state, info=batch.info)
-        if isinstance(logits, tuple):
-            dist = self.dist_fn(*logits)
+        # See https://pytorch.org/docs/stable/distributions.html
+        action_dist_input, hidden = self.actor(batch.obs, state=state, info=batch.info)
+        if isinstance(action_dist_input, tuple):
+            dist = self.dist_fn(*action_dist_input)
         else:
-            dist = self.dist_fn(logits)
+            dist = self.dist_fn(action_dist_input)
 
         # in this case, the dist is unused!
         if self.deterministic_eval and not self.training:
-            act = self._get_deterministic_action(logits)
+            act = self._get_deterministic_action(action_dist_input)
```

Comment:
@MischaPanch I am thinking about changing this line to:

```python
# Instead of using the action_dist_input use dist
act = self._get_deterministic_action(dist)
# and then use dist.mean or torch.argmax(dist.probs)
```
I think is much more compact if the `act` variable comes *always* from a distribution and not from raw output (i guess simply unnormalized probabilities or mean, std_dev).

The code is also more clear, if the action is deterministic i take either the mean or the most probable action from the distribution and if not i just sample from the distribution. But then it is clear that in both cases the action comes from the distribution.